### PR TITLE
Initial implementation for password reset and general profile and some…

### PR DESCRIPTION
**Pull request by pankajssp**
Initial implementation for password reset and general profile and some bug fixes from last commit.

### CODE REVIEW

**Comment by koushikkm** [comment]
Done with my review. I have not reviewed the javascript and template part.

**Comment by koushikkm** [comment]
you can handle the review comments by checking in over this pull request or you can merge it and then handle them in next request. But its better to resolve them in the same pull request, then we will not miss any comment.

File: account/models.py [hunk 1]
```diff
@@ -62,9 +63,33 @@ class BasicUser(AbstractBaseUser):
 
     #this tells django which is the unique field here, by default it is username 
     USERNAME_FIELD = 'email'
+    is_active = models.BooleanField(default=True)
     
     objects = MyUserManager()
 
     def __str__(self):
         return self.email
+
```
Comment (pankajssp) [inline account/models.py]: Put in TODO list would be done.

File: account/models.py [hunk 2]
```diff
@@ -62,9 +63,33 @@ class BasicUser(AbstractBaseUser):
...
     USERNAME_FIELD = 'email'
+    is_active = models.BooleanField(default=True)
     
     objects = MyUserManager()
 
     def __str__(self):
         return self.email
+        
+#Model defining Extended Profile of a User
+class UserProfile(models.Model):
```
Comment (pankajssp) [inline account/models.py]: Put in TODO list would be done.

File: account/models.py [hunk 3]
```diff
@@ -62,9 +63,33 @@ class BasicUser(AbstractBaseUser):
...
+	GENDER = (('M', 'Male'),('F', 'Female'),('O','Others'))
+	OCCUPN = (('S','Student'),('P','Professional'),('O','Others')) 
+	COUNTRY = (('I','India'),('O','Others'))
+	STATE = (('K','Karnatka'),('O','Others'))
+	CITY = (('B','Bangalore'),('O','Others'))
+	
+	t_user = models.ForeignKey(BasicUser, on_delete=models.CASCADE, to_field="email")
```
Comment (pankajssp) [inline account/models.py]: Put in TODO list would be done.

File: account/models.py [hunk 4]
```diff
@@ -62,9 +63,33 @@ class BasicUser(AbstractBaseUser):
...
+	gender = models.CharField("Gender", max_length=20, default="others", choices=GENDER, blank=True)
+	d_o_b = models.DateField("Date Of Birth", blank=True, help_text="Please enter in the Format\
```
Comment (pankajssp) [inline account/models.py]: Put in TODO list would be done.

File: account/models.py [hunk 5]
```diff
@@ -62,9 +63,33 @@ class BasicUser(AbstractBaseUser):
...
+							dd/mm/yyyy")
+	occupation = models.CharField("occupation", max_length=20, default="others",choices=OCCUPN, blank = True)
+	organization_name = models.CharField("organization name", max_length=500, default="None", blank=True)
+	company_name = models.CharField("company name", max_length=500, default="None", blank=True)
```
Comment (pankajssp) [inline account/models.py]: Put in TODO list would be done.

File: account/models.py [hunk 6]
```diff
@@ -62,9 +63,33 @@ class BasicUser(AbstractBaseUser):
...
+	designation = models.CharField("designation", max_length=500, default="None", blank=True)
+	country = models.CharField("country", max_length=500, default="others",choices=COUNTRY, blank=True)
+	state = models.CharField("state", max_length=500, default="others", choices=STATE, blank=True)
+	city = models.CharField("city", max_length=500, default="others",choices=CITY, blank=True)
+	billing_method = models.CharField("billing option", max_length=500, default="None", blank=True)
+	last_logn = models.DateTimeField("last login", blank=True)
```
Comment (pankajssp) [inline account/models.py]: Put in TODO list would be done.

File: account/views.py [hunk 7]
```diff
@@ -1,68 +1,141 @@
...
 #global variable used to communicate state between
 #python and JS files
 stateVar = 0
 
+"""
+TODO List for profile change:
+1>Error handling on net connection gone
+"""
+
+@login_required(login_url='account/login.html')
```
Comment (pankajssp) [inline account/views.py]: Fixed

File: account/views.py [hunk 8]
```diff
@@ -1,68 +1,141 @@
...
+	
 	#pkcomment form.is_valid will call form.clean automatically.
 	#will ensure calling of Authenticate() method
 	if request.method == "POST" and form.is_valid():
 		t_user = form.save()		
-		tlogin(request, t_user)
-		return render(request, 'account/mypage.html', {'stateVar':stateVar})	
-	stateVar = 5
-	return render(request, 'account/login.html', {"form": form, 'stateVar':stateVar})                
+		tlogin(request, t_user)
```
Comment (pankajssp) [inline account/views.py]: Fixed

File: account/views.py [hunk 9]
```diff
@@ -1,68 +1,141 @@
...
+def logout(request):	
 	global stateVar
 	stateVar = 5
-	logout(request)
+	tlogout(request)
 	return render(request, 'account/thome.html', {'stateVar':stateVar}) 
         
-def register(request):
-   
+def register(request):
```
Comment (pankajssp) [inline account/views.py]: Fixed

File: account/forms.py [hunk 10]
```diff
@@ -39,6 +41,15 @@ class Meta:
...
-        fields = ['full_name', 'mobile_number', 'email', 'passwd']
-        widgets = {'passwd': forms.PasswordInput()}
+        fields = ['full_name', 'mobile_number', 'email', 'password'] 
+        widgets = {'password': forms.PasswordInput()}
+
+class UserProfileForm(ModelForm):
+    class Meta:
+        model = UserProfile       
+       # fields = '__all__' 
+        fields = ['gender', 'd_o_b', 'occupation', 'organization_name', 'company_name',\
```
Comment (pankajssp) [inline account/forms.py]: Fixed

File: account/forms.py [hunk 11]
```diff
@@ -39,6 +41,15 @@ class Meta:
...
+        		 'designation', 'country', 'state', 'city', 'billing_method', 'last_logn']
```
Comment (pankajssp) [inline account/forms.py]: Fixed

**Comment by pankajssp** [comment]
Replied for review comments. Fixed item would be available in the next commits.

**Pull request merged**